lib/src/main/java/io/ably/lib/http/AsyncHttpScheduler.java

```diff
@@ -6,27 +6,49 @@
 import java.util.concurrent.ThreadPoolExecutor;
 import java.util.concurrent.TimeUnit;
 
+import io.ably.lib.util.Log;
+
 /**
  * A HttpScheduler that uses a thread pool to run HTTP operations.
  */
-public class AsyncHttpScheduler extends HttpScheduler<ThreadPoolExecutor> {
+public class AsyncHttpScheduler extends HttpScheduler {
     public AsyncHttpScheduler(HttpCore httpCore, ClientOptions options) {
-        super(httpCore, new ThreadPoolExecutor(options.asyncHttpThreadpoolSize, options.asyncHttpThreadpoolSize, KEEP_ALIVE_TIME, TimeUnit.MILLISECONDS, new LinkedBlockingQueue<Runnable>()));
-        executor.allowsCoreThreadTimeOut();
-    }
-
-    public void dispose() {
-        ThreadPoolExecutor threadPoolExecutor = executor;
-        threadPoolExecutor.shutdown();
-        try {
-            threadPoolExecutor.awaitTermination(SHUTDOWN_TIME, TimeUnit.MILLISECONDS);
-        } catch (InterruptedException e) {
-            threadPoolExecutor.shutdownNow();
-        }
+        super(httpCore, new WrappedExecutor(options));
     }
 
     private static final long KEEP_ALIVE_TIME = 2000L;
-    private static final long SHUTDOWN_TIME = 5000L;
 
     protected static final String TAG = AsyncHttpScheduler.class.getName();
+
+    private static class WrappedExecutor implements CloseableExecutor {
```

Comment:
I think the name of this class can be improved, now we only know that the executor is wrapped, but we don't know why. I'd like to include the reason behind the wrapper in this class name as well. Maybe something like `ClosableThreadPoolExecutor` or something similar? :thinking:

Reply:
That's a good point. I was struggling with naming. I shall adopt your suggestion. Thanks!

Reply:
https://github.com/ably/ably-java/pull/807/commits/dce84ef20d89edbd663e6996371257a35eec6549